packages/build/tests/plugins_list/snapshots/tests.js.md

```diff
@@ -3304,3 +3304,65 @@ Generated by [AVA](https://avajs.dev).
     ────────────────────────────────────────────────────────────────␊
     ␊
     (Netlify Build completed in 1ms)`
+
+## Errors if unsupported Next.js Runtime plugin version is installed in package.json
+
+> Snapshot 1
+
+    `␊
+    ────────────────────────────────────────────────────────────────␊
+      Netlify Build                                                 ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      featureFlags:␊
+        - plugins_break_builds_with_unsupported_plugin_versions␊
+      repositoryRoot: packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json␊
+      testOpts:␊
+        pluginsListUrl: /test/socket␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json␊
+    ␊
+    > Config file␊
+      packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json/netlify.toml␊
+    ␊
+    > Resolved config␊
+      build:␊
+        publish: packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json␊
+        publishOrigin: default␊
+      plugins:␊
+        - inputs: {}␊
+          origin: config␊
+          package: '@netlify/plugin-nextjs'␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    > Available plugins␊
+    ␊
+    > Using Next.js Runtime - v1.0.0␊
```

Comment:
I see the tests snapshot is updated and tests are passing, but it's not clear to me how this condition

`semver.satisfies(version, '4.0.0 - 4.25.0', { includePrerelease: true })` works in the context of the test, if the test is saying `1.0.0.` is invalid. Maybe I'm missing something in regards to how tests are set up for the build project.

Reply:
This is because version numbers are [normalized](https://github.com/netlify/build/blob/main/packages/testing/src/normalize.ts#L115) in snapshot outputs for example [here](https://github.com/netlify/build/blob/main/packages/build/tests/plugins_list/tests.js#L31) to 1.0.0 each time for stability. Pulling in @netlify/dev-workflow-pod-build-services to sanity check my claims.

Changing version in the fixtures [here](https://github.com/netlify/build/blob/fix/jackie-fail-build-on-unsupported-version-updates/packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json/node_modules/%40netlify/plugin-nextjs/package.json#L3) and [here](https://github.com/netlify/build/blob/fix/jackie-fail-build-on-unsupported-version-updates/packages/build/tests/plugins_list/fixtures/plugins_unsupported_version_package_json/package.json#L9) however breaks the snapshots as expected.

Reply:
Awesome. Thanks for the explanation @jackiewmacharia.